tests/unit/config/test_configfiles.py

```diff
@@ -233,6 +233,22 @@ def test_bindings_default(self, yaml, autoconfig):
         data = autoconfig.read()
         assert 'bindings.default' not in data
 
+    @pytest.mark.parametrize('show',
+                             [True, False, 'always', 'never', 'pinned'])
+    def test_tabs_favicons_show(self, yaml, autoconfig, show):
+        """Tests for migration of tabs.favicons.show."""
+        autoconfig.write({'tabs.favicons.show': {'global': show}})
+
+        yaml.load()
+        yaml._save()
+
+        data = autoconfig.read()
+        if isinstance(show, bool):
+            when = 'always' if show else 'never'
+        else:
+            when = show
```

Comment:
It often pays off to be a bit more explicit about expected values (and it's also nice "documentation") - I changed that (also for the test above) in 32568a6da44468a03a2e08b1a0cbe8c4a913c3ca.